File: src/Dfe.Complete.Domain/Entities/LocalAuthority.cs
using System;
using System.Collections.Generic;

namespace Dfe.Complete.Infrastructure.Models;

public partial class LocalAuthority
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public string Code { get; set; } = null!;

    public string Address1 { get; set; } = null!;

    public string? Address2 { get; set; }

    public string? Address3 { get; set; }

    public string? AddressTown { get; set; }

    public string? AddressCounty { get; set; }

    public string AddressPostcode { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: src/Tests/Dfe.Complete.CypressTests/cypress/pages/service-support/localAuthority.ts
import BasePage from "cypress/pages/basePage";

class LocalAuthority extends BasePage {
    protected readonly nameId = "Name";
    protected readonly codeId = "Code";
    protected readonly address1Id = "Address1";
    protected readonly address2Id = "Address2";
    protected readonly address3Id = "Address3";
    protected readonly addressTownId = "AddressTown";
    protected readonly addressCountyId = "AddressCounty";
    protected readonly addressPostcodeId = "AddressPostcode";

    protected readonly dcsPositionId = "Title";
    protected readonly dcsNameId = "ContactName";
    protected readonly dcsEmailId = "Email";
    protected readonly dcsPhoneId = "Phone";

    saveAndReturn() {
        this.clickButton("Save and return");
        return this;
    }
}

export default LocalAuthority;
